### DISCUSSION TITLE
USB interface peripherals

### DISCUSSION THREAD

**Discussion by LouDnl** [vossstef/tang_nano_20k_c64#82] [General]
Hi, what would be needed to interface peripherals that can be accessed via an USB-CDC device?
For example I am making a USB SID device that can host 2 real SIDs. Is it possible to disable the fpga sid code and send sid addr/data over cdc?

**Comment by vossstef** [comment]
@LouDnl ,
concerning possibility to do such in the firmware you better ask @harbaum . I developed the core and mainly use the firmware as such with little changes just in the OSD Menu.

**Comment by harbaum** [comment]
You'd need to write a USB driver for your device for the MCU side of things (e.g. the bl616, esp32 or rp2040). Then you need to implement an interface between the MCU with it's USB driver and the fpga to forward these requests.

This is sure possible. The question is whether the size of the possible audience justifies this effort.

**Reply by LouDnl** [reply]
@harbaum It wouldn't have to be more then using the serial uart already implemented `RS232 Serial Interface [VIC-1011](http://www.zimmers.net/cbmpics/xother.html) or [UP9600](https://www.pagetable.com/?p=1656) mode to Tang onboard USB-C serial port.`
Since there already is some kind of serial uart implementation it (i think) would be route sid address/data to the uart?
The protocol is simple and could be easily implemented? I have 0 fpga knowledge but with some pointers I might be able to create some poc?
